extensions/amp-analytics/0.1/events.js

```diff
@@ -1638,21 +1633,43 @@ export class VisibilityTracker extends EventTracker {
     return 'onpagehide' in this.root.ampdoc.win;
   }
 
+  /**
+   * Sets the default waitForSpec based upon if the element
+   * passed in is an AMP element or not.
+   * @param {string|undefined} waitForSpec
+   * @param {!Element} element
+   * @return {string}
+   */
+  setDefaultWaitForElement_(waitForSpec, element) {
+    if (!waitForSpec) {
+      waitForSpec = isAmpElement(element) ? 'ini-load' : 'none';
+    }
+    return waitForSpec;
+  }
+
   /**
    * @param {string|undefined} waitForSpec
    * @param {Element=} opt_element
    * @return {?Promise}
    * @visibleForTesting
    */
   getReadyPromise(waitForSpec, opt_element) {
+    if (opt_element) {
+      waitForSpec = this.setDefaultWaitForElement_(waitForSpec, opt_element);
```

Comment:
`waitForSpec` will never be `null` after `setDefaultWaitForElement_`. Do you mind clean up the `waitForSpec` default assignment logic a little bit. To get rid of `null`, also to have all default assignment logic moved to within `getReadyPromise` function. Thanks

Reply:
`null` a `'none'` should essentially be the same. You can switch to 
```
 readyPromiseWaitForSpec = waitForSpec || (selector ? 'ini-load' : 'none'); 
```

Reply:
what about
```
    if (opt_element) {
      if (!isAmpElement(element)) {
        userAssert(!waitForSpec || waitForSpec == 'none', 'error msg');
      } else {
        waitForSpec = waitForSpec || 'ini-load';
      }
    }
    
    if (!waitForSpec || waitForSpec == 'none') {
      return null;
    }

    const trackerAllowlist = getTrackerTypesForParentType('visible');
    userAssert(trackerAllowlist[waitForSpec] !== undefined,
      'waitFor value %s not supported',
      waitForSpec
    ); .......
```

Reply:
Looks like we're inline the `setDefaultWaitForElement_()` and we're adding a more detailed error message. I like it.